### ISSUE TITLE
Question: loss terms of VQ-VAE are confusing

### ISSUE THREAD

**Issue by AranKomat** [tensorflow/tensor2tensor#852]
### Description

If I understand correctly, for each iteration one should perform an EM update and a gradient update w.r.t. the loss consisting of two terms: latent prediction term and reconstruction term. According to the default setting of transformer_vae.py, there are two loss terms losses["extra"] and losses["latent_pred"], which are set to `beta * tf.reduce_mean(tf.square(x - tf.stop_gradient(x_means)))` and `tf.reduce_mean((inputs_c - targets_c)**2) * 20` (see Line 391), respectively (see `ae_transformer_internal` method). But neither of them looks like the reconstruction loss, which I believe is a cross-entropy loss. Furthermore, losses["latent_pred"] here doesn't seem to appear in the original paper, and and losses["extra"] seems like the actual latent prediction loss that appeared in the paper. Am I supposed to define the cross-entropy reconstruction loss using the 'res' output of 'ae_transformer_internal' and add it to losses["extra"] above to get the valid loss function?
